### DISCUSSION TITLE
q-uploader issues on Android and iOS

### DISCUSSION THREAD

**Discussion by minuba-mdp** [quasarframework/quasar#12106] [General - Components / Directives / etc]
Hello Quasar Team.

I’m currently building a file uploader for both Android and iOS.

I have a problem with both platforms.

**iOS (iOS 15.1, iPhone SE):**
When I press the ‘add’ button on the q-uploader in my app and use the option ‘Take Photo or Video’ then switch to ‘Video’ the app crashes completely.

When using a browser on my iPhone and going to the q-uploader [docs](https://quasar.dev/vue-components/uploader#design) and using the component there, it does not crash.

**Android (Android 8.0.0, Huawei Y6 2018):**
When I press the ‘add’ button on the q-uploader in my app, I get no options but it opens the file storage.

When I use a browser on my Android mobile and go to the q-uploader [docs ](https://quasar.dev/vue-components/uploader#design) and use the component there, it gives me the options like on iOS.

So my question is; how can I fix these issues?

**Quasar version:** 1.17.1

**Screenshot of the template:**

![2022-01-19 16_01_09](https://user-images.githubusercontent.com/98037609/150157183-d629efc5-08ab-4593-9ba2-7ce7b999e1f4.png)

If you need any other specifications, please let me know.

Kind regards,
Michael

**Comment by pdanpdan** [comment]
Check `capture` and `accept` props
https://quasar.dev/vue-components/uploader#usage
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/capture

For the crash you need to check who is crashing / message

**Reply by minuba-mdp** [reply]
I tried using the `accept` prop, but without any luck, even if I specify only `image/*` (can't remember now if it's image or images).

I just don't understand how the exact same code can be different for iOS and Android? Does it have something to do with Android requiring `capture` and `accept` props?

I will try adding both props soon and get back with the result. Also, I will try to find the crash message.

Thanks for your answer, I hope using both props will fix it.

**Reply by minuba-mdp** [reply]
Hello @pdanpdan. 

I have implemented both props and made a build for Android. The same problem occurs. Any thoughts?

Below is the screenshot with the new props (I used `media_type` from this [page](https://www.w3schools.com/tags/att_input_accept.asp#attribute_values))
![billede](https://user-images.githubusercontent.com/98037609/150535365-0fa6914f-cc87-4c2c-8da6-5113271ef3a1.png)

Currently, I'm having difficulties getting the crash message on iOS. Since the emulator cannot use the camera, and when the app crashes on a device, the developer tool from Safari also closes. Any thoughts?

**Reply by pdanpdan** [reply]
`:accept="media_type"`

**Reply by minuba-mdp** [reply]
`media_type` is not a variable I have created, but a value for the `accept` prop, therefore I get a compile error when using `:`. I found `media_type` from the [link ](https://www.w3schools.com/tags/att_input_accept.asp) above.

`media_type` should contain any type of media, since the component I'm creating should be able to upload images, videos, documents, PDFs, and more.

**Reply by rstoenescu** [reply]
> `media_type` is not a variable I have created, but a value for the `accept` prop, therefore I get a compile error when using `:`. I found `media_type` from the [link ](https://www.w3schools.com/tags/att_input_accept.asp) above.
> 
> 
> 
> `media_type` should contain any type of media, since the component I'm creating should be able to upload images, videos, documents, PDFs, and more.

Use MDN instead of that. "media_type" is not a valid value and does not specify anything usable.

**Reply by minuba-mdp** [reply]
Hey @rstoenescu.

Thank you for your reply.

I'm not that familiar with the accept and capture props and I couldn't find documentation for MDN, would you mind explaining what MDN is or stands for?

I tried replacing `media_type` with `MDN`, with the same result.
![billede](https://user-images.githubusercontent.com/98037609/150741407-a6f1be48-6704-4223-946c-34742f08b34f.png)

**Reply by pdanpdan** [reply]
MDN is the thing you got a documentation link exactly at `capture` attribute in the first reply

**Reply by minuba-mdp** [reply]
Thanks @pdanpdan.

Any thought on why the change from `media_type` to `MDN` didn't make any difference?

Since I can't get any crash messages from my iOS device, I was wondering if it's possible to disable the ability to use video from the camera, but still, be able to upload videos already recorded?

**Reply by minuba-mdp** [reply]
I get the now... instead of using this [link](https://www.w3schools.com/tags/att_input_accept.asp) I should use this [link](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept).

I will be back after some reading.

**Reply by minuba-mdp** [reply]
Even when I set the `accept` prop to `image/*`
![billede](https://user-images.githubusercontent.com/98037609/150746314-207998a6-61e7-4685-b541-d99e5f27b326.png)

I will not show the option to open the camera as it does on your documentation page.
![billede](https://user-images.githubusercontent.com/98037609/150747120-2d0542f5-dccf-4ef5-ba77-eeb744feab0d.png)

Any thoughts?

**Reply by minuba-mdp** [reply]
Hey @pdanpdan and @rstoenescu, I have been wondering...

Have you ever heard of a problem like this before?

Could my issue be with the libraries that the QUploader uses and not the component itself?

Which libraries does the QUploader use to interact with the native camera and file storage?

Would it do any help getting my package.json/package-lock.json for versions of other libraries?

Thanks for your time helping me.

**Reply by minuba-mdp** [reply]
I haven't heard from you guys in some time, are you still looking into this issue, or have I been left on the floor?
